packages/react-reconciler/src/ReactFiberCommitWork.old.js

```diff
@@ -580,6 +587,9 @@ function commitHookEffectListMount(flags: HookFlags, finishedWork: Fiber) {
         }
 
         if (__DEV__) {
+          if ((effect.tag & HookInsertion) !== NoHookEffect) {
+            setInsideInsertionEffectDEV(false);
```

Comment:
I think a better place to put this would be here: https://github.com/facebook/react/blob/149b420f63903bcfb99455337a376c75384f86af/packages/react-reconciler/src/ReactFiberCommitWork.new.js#L1831

Reply:
This is inside `commitHookEffectListMount`, do you mean the code below should also be in here?

Reply:
Or I see, you mean wrap the actual call since we're passing `HookInsertion` flag so we assume only `useInsertionEffect` will run inside it?

Reply:
Hmm, there are 2 callsites passing `HookInsertion | HookHasEffect` to `commitHookEffectListMount` so I will have to repeat this logic twice, is that preferable? I put it inside because there was already DEV only logic in here

Reply:
Also, should we wrap the `unmount` too?